File: src/main/java/com/robwestz/figmatobackend/ui/FigmaToolWindowFactory.java
package com.robwestz.figmatobackend.ui;

import com.intellij.openapi.project.Project;
import com.intellij.openapi.wm.ToolWindow;
import com.intellij.openapi.wm.ToolWindowFactory;
import com.intellij.ui.content.Content;
import com.intellij.ui.content.ContentFactory;
import org.jetbrains.annotations.NotNull;

public class FigmaToolWindowFactory implements ToolWindowFactory {
    
    @Override
    public void createToolWindowContent(@NotNull Project project, @NotNull ToolWindow toolWindow) {
        FigmaToolWindowPanel toolWindowPanel = new FigmaToolWindowPanel(project);
        ContentFactory contentFactory = ContentFactory.getInstance();
        Content content = contentFactory.createContent(toolWindowPanel.getContent(), "", false);
        toolWindow.getContentManager().addContent(content);
    }
}

File: src/main/kotlin/com/minorg/figmabridge/toolwindow/FigmaToolWindowFactory.kt
package com.minorg.figmabridge.toolwindow

import com.intellij.openapi.project.Project
import com.intellij.openapi.wm.ToolWindow
import com.intellij.openapi.wm.ToolWindowFactory
import com.intellij.ui.content.ContentFactory

/**
 * Denna klass anropas av PyCharm när användaren klickar på "Figma Bridge"-fliken.
 * Den ansvarar för att instansiera och bygga själva UI-panelen.
 */
class FigmaToolWindowFactory : ToolWindowFactory {

    override fun createToolWindowContent(project: Project, toolWindow: ToolWindow) {
        // 1. Skapa vår anpassade UI-panel
        val figmaPanel = FigmaToolWindowPanel()

        // 2. Hämta fabriken för att skapa "innehåll"
        val contentFactory = ContentFactory.getInstance()

        // 3. Skapa ett nytt innehållsobjekt med vår panel
        val content = contentFactory.createContent(figmaPanel, "", false)

        // 4. Lägg till det skapade innehållet i verktygsfönstret
        toolWindow.contentManager.addContent(content)
    }
}
